e2e/test/prometheus.spec.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2024 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+import { afterAll, beforeAll, describe, expect, test } from "@jest/globals";
+import * as net from 'net';
+import { closeForward, getForward } from './forward';
+
+const dev = process.env.DEV == 'true';
+
+describe('Prometheus and Alertmanager', () => {
+  let prometheusProxy: { server: net.Server, url: string };
+  let alertmanagerProxy: { server: net.Server, url: string };
+
+  beforeAll(async () => {
+    prometheusProxy = await getForward('kube-prometheus-stack-prometheus', 'monitoring', 9090);
+    alertmanagerProxy  = await getForward('kube-prometheus-stack-alertmanager', 'monitoring', 9093);
+  })
+
+  afterAll(async () => {
+    await closeForward(prometheusProxy.server)
+    await closeForward(alertmanagerProxy.server)
+  });
+  
+  test('alert manager service should be responsive via the internal service address', async () => {
+    const response = await fetch(`${alertmanagerProxy.url}`);
+    expect(response.status).toBe(200);
+  });
+
+  test('alert manager should be firing watchdog alert', async () => {
+    // fetch active alerts with alertname="Watchdog"
+    const response = await fetch(`${alertmanagerProxy.url}/api/v2/alerts/groups?filter=alertname%3D%22Watchdog%22&silenced=false&inhibited=false&active=true`);
+    expect(response.status).toBe(200);
+
+    const body = await response.json() as unknown as [{ alerts: { status: { state: string } }[] }];
+    expect(body[0]).toBeDefined();
+    expect(body[0].alerts[0].status.state).toEqual('active');
+  });
+
+  test('prometheus web ui should be responsive via the internal service address', async () => {
+    const response = await fetch(`${prometheusProxy.url}`);
+    expect(response.status).toBe(200);
+  });
+
+  test('all prometheus targets should be up', async () => {
+    const response = await fetch(`${prometheusProxy.url}/api/v1/targets`);
+    expect(response.status).toBe(200);
+
+    const body = await response.json() as unknown as { data: { activeTargets: any[] } };
+    expect(body.data).toBeDefined();
+    expect(body.data.activeTargets.length).toBeGreaterThan(0);
+
+    // filter out unknown because test runner pod is unknown, probably can filter out specific pod
+    const failedTargets = body.data.activeTargets.filter(target => target.health !== 'up' && target.health !== 'unknown');
```

Comment:
I think `unknown` refers to the target not having been scraped yet ([ref](https://training.promlabs.com/training/monitoring-and-debugging-prometheus/web-status-pages/target-scrape-status/)). I think this might be okay but would be a bit worried we'd miss something by filtering those out, if something _would fail_ once scraped? Not really sure what to do here, I don't think we strictly want to fail this, but if we could throw a warning message with any unknown targets?

Reply:
Updated to warn if unknown targets are encountered.  I think the only real fix for this would be adding a synthetic delay to wait for all targets to be healthy.